domain/resource/service/resource.go

```diff
@@ -38,18 +40,32 @@ type State interface {
 	// GetResource returns the identified resource.
 	GetResource(ctx context.Context, resourceUUID coreresource.UUID) (resource.Resource, error)
 
-	// SetResource adds the resource to blob storage and updates the metadata.
-	SetResource(ctx context.Context, config resource.SetResourceArgs) (resource.Resource, error)
+	// StoreResource records a stored resource along with who retrieved it.
+	// Returns [resourceerrors.ResourceNotFound] if the resource UUID cannot be found.
+	// Returns [resourceerrors.StoredResourceNotFound] if the stored resource at the
+	// storageID cannot be found.
+	// Returns [resourceerrors.ResourceAlreadyStored] if the resource is already
+	// associated with a stored resource blob.
+	// Returns [resourceerrors.RetrievedByTypeNotValid] if the retrieved by type is
+	// invalid.
+	StoreResource(
+		ctx context.Context,
+		resourceUUID coreresource.UUID,
+		storageUUID coreresourcestore.ID,
+		retrievedBy string,
+		retrievedByType resource.RetrievedByType,
+		incrementCharmModifiedVersion bool,
```

Comment:
Very true, have changed. Though I've left the state one as is because the arguments are different and it would mean that we had to put two `StoreResource` looking structs in the domain types.